packages/notebook/src/widget.ts

```diff
@@ -529,13 +533,15 @@ export class StaticNotebook extends WindowedList {
     }
     this._updateMimetype();
     const cells = newValue.cells;
-    if (!cells.length) {
+    const collab = PageConfig.getOption('collaborative');
+    if ((!collab || collab == 'false') && !cells.length) {
```

Comment:
```suggestion
    const collab = (PageConfig.getOption('collaborative') ?? '').toLowerCase() === 'true';
    if (!collab && !cells.length) {
```

Reply:
Actually would it be better to do this if only a single user is connected? This will allow the behavior to be consistent on deployed infrastructure that run in collaborative mode - flag that will likely be drop in the future.

Reply:
I don't think checking for more than one user is safe.
I believe we could have a case where one client disconnected for a moment, and then we added an empty cell from both clients.

Reply:
Because this is a special case that doesn't come from an event in the sharedModel.
This method is called when changing the NotebookModel in the StaticNotebook.

Reply:
@dmonad if I am understanding this, if there are 3 cells left and 3 different users delete 1 cell at the same time, their local client won't see them as deleting the last cell, so no new one would be created if that logic is in delete cell.

Reply:
@ellisonbg 

That's correct. There's a slight chance that users end up with no cells at all if multiple concurrent actions lead to an empty notebook. While undesirable, users can still add a new cell manually when they detect an empty notebook (which is technically not a problem).

On the other hand, if all clients add a new cell whenever they detect an empty notebook, they will end up with `N` empty cells whenever anyone deletes the last cell. Another undesirable state, but it is more likely to happen in practice.

(**) Hence my proposal is to only add an empty cell when the last cell is deleted (in `sharedModel.deleteCell`). If you do this in the same transaction, then the notebook will never be empty (unless users delete cells simultaneously). 

To guard against simultaneous deletion, you could use the server to detect an empty notebook to add a new cell. 

In any case (**) should be implemented. Otherwise, users will have an empty notebook until the server responds. The idea of shared-models is also that they work independently from the server. JupyterLite, for example, works peer-to-peer using y-webrtc and doesn't have a central server for coordination.

@hbcarlos
Yeah, I didn't see that. If (**) and the server approach is implemented, this special case shouldn't be necessary. Personally, I'd try to avoid special conditions unless necessary. But it also doesn't seem to hurt.